web/source/osk/visualKeyboard.ts

```diff
@@ -1055,6 +1055,11 @@ namespace com.keyman.osk {
       //        update their geometries to the actual display values, and use the results here.
       let touchKbdPos = this.getTouchCoordinatesOnKeyboard(touch);
       let layerGroup = this.kbdDiv.firstChild as HTMLDivElement;  // Always has proper dimensions, unlike kbdDiv itself.
+      let width = layerGroup.offsetWidth, height = layerGroup.offsetHeight;
+      // Prevent NaN breakages.
+      if(!width || !height) {
+        return null;
+      }
```

Comment:
Is this a known issue that is being addressed here?

Reply:
It showed up sometimes during my debugging runs, as I tested this one more proactively inside of iOS, rather than on the bootstrapping page.  Given that associated issues don't show up in Sentry, I'm guessing that it's something that arises due to the combination of breakpoint use and running within a `WKWebView`.  Unless Sentry's failing to capture it somehow?

So... it makes iOS testing a bit more stable.

----

Keep in mind, this scenario _should_ only arise if `layerGroup` isn't visible, but this method triggers on keystrokes _against `layerGroup`'s child elements_.  I, the user, am triggering keystrokes against the (visible) layer, yet the layer group's being reported invisible somehow?

I could make an issue about investigating this, I suppose.

Reply:
Android isn't generating any associated errors for its embedded KMW, which would be expected if it weren't iOS-dependent.

iOS should also report unhandled console errors, so the fact that it _also_ hasn't had anything show up, suggests it's something that currently requires an Xcode debugging session.  I've tried reproducing sequences that caused the console errors I saw on my physical device, with error reporting on, and nothing shows up in Sentry.

Reply:
Maaaybe I should run a test with Android + Android Studio debugging, to be sure it wouldn't arise there either.  Not sure if that info would be particularly useful at the moment though.

Reply:
OK. It may be worth investigating one day if you are feeling bored!